src/main/java/uk/gov/justice/digital/delius/jpa/standard/repository/ContactDateRepositoryImpl.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.digital.delius.jpa.standard.repository;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.support.PageableExecutionUtils;
+import org.springframework.stereotype.Repository;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Contact;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.time.LocalDate;
+
+import static org.springframework.data.jpa.repository.query.QueryUtils.toOrders;
+
+/**
+ * Repository of distinct contact dates.
+ */
+@Repository
+public class ContactDateRepositoryImpl implements ContactDateRepository {
```

Comment:
I guess the below couldn't be written as a JPA query since you need the specification has a starting point.

I was wondering if PagingAndSortingRepository would help here at all, but can't think how

Reply:
Yeah so I did a fair bit of investigating on this as I also wasn't happy with going bare JPA. As you say the problem is that there is no way to build JPQL and then apply a Specification and there's no overload of `findAll` that takes a Specification whilst providing some way of modifying the select query. Key for this repository is `distinct CONTACT_DATE`. It is I admit an unusual case and the resulting raw JPA is pretty compact so in the end I was happy to leave it in.